accessor.go

```diff
@@ -27,6 +31,12 @@ type ShardAccessor struct {
 	data  mount.Reader
 	idx   index.Index
 	shard *Shard
+
+	// mmapr is an optional mmap.ReaderAt. It will be non-nil if the mount
+	// has been mmapped because the mount.Reader was an underlying *os.File,
+	// and an mmap-backed accessor was requested (e.g. Blockstore).
+	lk    sync.Mutex
```

Comment:
Actually, this lock might not even be necessary because ShardAccessors aren't thread-safe because by definition because they operate on a reader.